packages/block-editor/src/store/selectors.js

```diff
@@ -2348,6 +2348,34 @@ export function __experimentalGetDirectInsertBlock(
 	return getDirectInsertBlock( state, rootClientId );
 }
 
+const parsedPatternCache = new WeakMap();
+
+function parsePattern( pattern ) {
+	const blocks = parse( pattern.content, {
+		__unstableSkipMigrationLogs: true,
+	} );
+	if ( blocks.length === 1 ) {
+		blocks[ 0 ].attributes = {
+			...blocks[ 0 ].attributes,
+			metadata: {
+				...( blocks[ 0 ].attributes.metadata || {} ),
+				categories: pattern.categories,
+				patternName: pattern.name,
+				name: blocks[ 0 ].attributes.metadata?.name || pattern.title,
+			},
+		};
+	}
+	return {
+		...pattern,
+		blocks,
+	};
+}
+
+function getParsedPattern( pattern ) {
+	const parsedPattern = parsedPatternCache.has( pattern );
+	return parsedPattern ? parsedPattern : parsePattern( pattern );
+}
+
 export const __experimentalGetParsedPattern = createRegistrySelector(
```

Comment:
To do, maybe in a follow-up: deprecate this selector and update all uses.

Reply:
Actually, I just removed the memoization and it seems fine to leave for the remaining use case (pattern block).